components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -481,6 +508,91 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
   [self.view setNeedsLayout];
 }
 
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^_Nullable)(BOOL))completion {
+  CGSize newPreferredContentSize =
+      CGSizeMake(CGRectGetWidth(self.view.bounds), preferredContentHeight);
+  if ([self isMobileLandscape]) {
+    self.contentViewController.preferredContentSize = newPreferredContentSize;
+    return;
+  }
+  CGPoint contentOffset =
+      [self calculateContentOffsetWithPreferredContentHeight:preferredContentHeight];
+  CGFloat precentageOfFullScreen =
+      [self precentageOfFullScreenWithPreferredContentHeight:preferredContentHeight];
+  BOOL shouldSetContentHeight = NO;
+  if (_contentVCPreferredContentSizeHeightCached > preferredContentHeight) {
+    shouldSetContentHeight = YES;
+  } else {
+    if (_contentVCPreferredContentSizeHeightCached != preferredContentHeight) {
```

Comment:
If `_contentVCPreferredContentSizeHeightCached` is the same as the new height then we simply animate. If they are different and we are changing to a new `preferredContentSize` once we set that value (which is required so we don't see the scrim) the content will _jump_. In order to prevent this jump we need to set the contentOffset of the scroll view to account for this. We additionally need to set the content offset that we are animating to account for this change.